server/common/mail/templates/index.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const pupa = require('pupa');
+const base64Img = require('base64-img');
+const mjml2html = require('mjml');
+const iconPath = '/assets/shopping.png';
+
+function readTemplate(templatePath, params) {
+  const template = fs.readFileSync(path.join(__dirname, templatePath), 'utf8');
+  const icon = { icon: base64Img.base64Sync(path.join(__dirname, iconPath)) };
```

Comment:
In order to create data uri you need to get file mimetype and base64 encoded contents. To resolve mimetype use [`file-type`](https://npm.im/file-type) that is already added to project deps and for getting base64 encoded contents use `Buffer#toString` with `base64` param 💡 
To combine those two together use: [`create-data-uri`](https://npm.im/create-data-uri)

```js
const createDataUri = require('create-data-uri');
const fileType = require('file-type');

const buf = fs.readFileSync(...);
const { mime } = fileType(buf) || {};
const dataUri = createDataUri(mime, buf.toString('base64'));
```